packages/mui-system/src/cssVars/useCurrentColorScheme.ts

```diff
@@ -0,0 +1,290 @@
+  supportedColorSchemes: Array<SupportedColorScheme>;
+  defaultMode?: Mode;
+  modeStorageKey?: string;
+  colorSchemeStorageKey?: string;
+}): Result<SupportedColorScheme> {
+  const {
+    defaultMode = 'day',
+    defaultDayColorScheme,
+    defaultNightColorScheme,
+    supportedColorSchemes = [],
+    modeStorageKey = DEFAULT_MODE_STORAGE_KEY,
+    colorSchemeStorageKey = DEFAULT_COLOR_SCHEME_STORAGE_KEY,
+  } = options;
+
+  const joinedColorSchemes = supportedColorSchemes.join(',');
+
+  const [state, setState] = React.useState(() => {
+    const initialMode = resolveValue(modeStorageKey, defaultMode);
+    return {
+      mode: initialMode,
+      systemMode: getSystemMode(initialMode),
+      dayColorScheme: resolveValue(`${colorSchemeStorageKey}-day`) || defaultDayColorScheme,
+      nightColorScheme: resolveValue(`${colorSchemeStorageKey}-night`) || defaultNightColorScheme,
+    } as State<SupportedColorScheme>;
+  });
+
+  const colorScheme = getColorScheme(state);
+
+  const setMode: Result<SupportedColorScheme>['setMode'] = React.useCallback(
+    (mode) => {
+      setState((currentState) => {
+        const newMode = !mode ? defaultMode : mode;
+        if (typeof localStorage !== 'undefined') {
+          localStorage.setItem(modeStorageKey, newMode);
+        }
+        return {
+          ...currentState,
+          mode: newMode,
+          systemMode: getSystemMode(newMode),
+        };
+      });
+    },
+    [modeStorageKey, defaultMode],
+  );
+
+  const setColorScheme: Result<SupportedColorScheme>['setColorScheme'] = React.useCallback(
+    (value) => {
+      if (!value || typeof value === 'string') {
+        if (value && !supportedColorSchemes.includes(value)) {
+          console.error(`\`${value}\` does not exist in \`theme.colorSchemes\`.`);
+        } else {
+          setState((currentState) => {
+            const newState = { ...currentState };
+            if (!value) {
+              // reset to default color scheme
+              newState.dayColorScheme = defaultDayColorScheme;
+              newState.nightColorScheme = defaultNightColorScheme;
+              return newState;
+            }
+            processState(currentState, (mode) => {
+              localStorage.setItem(`${colorSchemeStorageKey}-${mode}`, value);
+              if (mode === 'day') {
+                newState.dayColorScheme = value;
+              }
+              if (mode === 'night') {
+                newState.nightColorScheme = value;
+              }
+            });
+            return newState;
+          });
+        }
+      } else if (
+        (value.day && !supportedColorSchemes.includes(value.day)) ||
+        (value.night && !supportedColorSchemes.includes(value.night))
+      ) {
+        console.error(`\`${value}\` does not exist in \`theme.colorSchemes\`.`);
+      } else {
+        setState((currentState) => {
+          const newState = { ...currentState };
+          if (value.day || value.day === null) {
```

Comment:
What if the new `value.day` is undefined? If it was previously set to something else, you won't update it no?

Reply:
`undefined` is neglect. I think we should support only `null` because it is intentional that the user wants to reset to the application default. local storage also returns `null` when the key is deleted.

Reply:
another reason that `undefined` is neglect because developers can do this:

```js
setColorScheme({ light: '...' })
// or
setColorScheme({ dark: '...' })
```

Reply:
Yeah that makes sense!